File: android/src/main/java/com/placepicker/PlacePickerPlugin.java
package com.placepicker;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PlacePicker")
public class PlacePickerPlugin extends Plugin {

    private PlacePicker implementation = null;

    @PluginMethod
    public void pickPlace(PluginCall call) {
        if (implementation == null) {
            implementation = new PlacePicker(this.getActivity());
        }

        implementation.start(call);
    }
}

File: ios/Plugin/PlacePickerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(PlacePickerPlugin)
public class PlacePickerPlugin: CAPPlugin {
    private let implementation = PlacePicker()

    @objc func pickPlace(_ call: CAPPluginCall) {
        implementation.start(call: call)
    }
}
